### ISSUE TITLE
Cannot open device with 2CQ on certain machine configurations

### ISSUE THREAD

**Issue by esmalTT** [tenstorrent/tt-metal#12945]
### Summary

For some reason, opening a device with 2 command queues will crash deterministically on certain N300/T3K machine configurations:

```txt
No more available dispatch cores on device 1 to assign. Expand dispatch cores specified in core descriptor YAML
backtrace:
 --- /localdev/esmal/tt-metal/ttnn/ttnn/_ttnn.so(+0x46a7d0) [0x7fe7524c77d0]
 --- /localdev/esmal/tt-metal/build_Release/lib/libtt_metal.so(+0x12b281) [0x7fe751f30281]
 --- tt::tt_metal::Device::setup_tunnel_for_remote_devices()
 --- tt::tt_metal::Device::compile_command_queue_programs()
 --- tt::tt_metal::Device::init_command_queue_device()
 --- tt::DevicePool::initialize_device(tt::tt_metal::Device*) const
 --- tt::DevicePool::init_firmware_on_active_devices() const
 --- tt::DevicePool::initialize(std::__1::vector<int, std::__1::allocator<int>>, unsigned char, unsigned long, unsigned long, tt::tt_metal::DispatchCoreType, std::__1::vector<unsigned int, std::__1::allocator<unsigned int>> const&)
 --- tt::tt_metal::CreateDevice(int, unsigned char, unsigned long, unsigned long, tt::tt_metal::DispatchCoreType, std::__1::vector<unsigned int, std::__1::allocator<unsigned int>> const&)
```

I have seen the issue on with `yyz-lab-60` and `sjc-lab-t3004` when trying to use ethernet dispatch cores. 

### Steps to reproduce

On `main`, run the following `pytest` script:

```python
@pytest.mark.parametrize(
    "device_params", [{"num_command_queues": 2}], indirect=True
)
def test_thing(
    device,
):
    return
```

You can also reproduce it using C++:

```cpp
#include <algorithm>
#include <functional>
#include <random>
#include <cstdlib>
#include <filesystem>

#include "tt_metal/host_api.hpp"
#include "common/bfloat16.hpp"
#include "tt_metal/llrt/tt_memory.h"
#include "tt_metal/detail/kernel_cache.hpp"
#include "tt_metal/detail/tt_metal.hpp"

#include "tt_metal/impl/device/device.hpp"
#include "tt_metal/impl/kernels/kernel.hpp"

using namespace tt;
using namespace tt::tt_metal;
int main(int argc, char **argv) {
    const int num_cqs = 2;
    const auto dispatch_core_type = DispatchCoreType::ETH;
    auto *device = tt::tt_metal::CreateDevice(0, num_cqs, DEFAULT_L1_SMALL_SIZE, DEFAULT_TRACE_REGION_SIZE, dispatch_core_type);
    return 0;
}
```

**Comment by aliuTT** [comment]
The issue is because on `yyz-lab-60`, the boards are unharvested for a N300 config, causing runtime to pick up the wrong ethernet dispatch assignment. This is unsupported across the company, we should ask syseng to harvest this machine properly. I do have a fix to "support" unharvested n300s here: https://github.com/tenstorrent/tt-metal/commit/96f97f0f26d210b783372681c4889341bbc21169. If I don't see any weirdness in CI, I will push it to main.

**Comment by aliuTT** [comment]
Fixed on main at 1dbdc62
